source/grid/grid_tools.cc

```diff
@@ -1573,6 +1573,16 @@ namespace GridTools
       return false;
     }
 
+    template <int dim, typename Cell>
+    inline
+    double calculate_cell_radius(const Cell &cell)
+    {
+      double res = 0.;
+      for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
+        res = std::max(res, ( cell->vertex(v)).distance(cell->center()) );
```

Comment:
@bangerth i guess you are right, given that cell's center is always located on the largest diagonal.

center -- The center of an object is defined to be the average of the locations of the vertices https://www.dealii.org/developer/doxygen/deal.II/classTriaAccessor.html#aef91f5dac9db66b70c05ce467263241c

diameter -- The diameter of an object is computed to be the largest diagonal. https://www.dealii.org/developer/doxygen/deal.II/classTriaAccessor.html#a46f3c5bc796657168a0a9e884e8eb3de